### PR TITLE
controllers: simplify deepcopy calls

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

**Comment by zhangxiaoyu-zidif** [comment]
very cool and clean code.

File: pkg/controller/controller_utils.go [hunk 1]
```diff
@@ -991,14 +987,7 @@ func PatchNodeTaints(c clientset.Interface, nodeName string, oldNode *v1.Node, n
...
 	newTaints := newNode.Spec.Taints
-	objCopy, err := scheme.Scheme.DeepCopy(oldNode)
-	if err != nil {
-		return fmt.Errorf("failed to copy node object %#v: %v", oldNode, err)
-	}
-	newNode, ok := (objCopy).(*v1.Node)
-	if !ok {
-		return fmt.Errorf("failed to cast copy onto node object %#v: %v", newNode, err)
-	}
+	oldNode.DeepCopyInto(newNode)
```
Comment (sttts) [inline pkg/controller/controller_utils.go]: this was has wrong direction

Comment (sttts) [reply pkg/controller/controller_utils.go]: no, in fact it's not.

File: pkg/controller/controller_utils.go [hunk 2]
```diff
@@ -991,14 +987,7 @@ func PatchNodeTaints(c clientset.Interface, nodeName string, oldNode *v1.Node, n
 	}
 
 	newTaints := newNode.Spec.Taints
-	objCopy, err := scheme.Scheme.DeepCopy(oldNode)
-	if err != nil {
-		return fmt.Errorf("failed to copy node object %#v: %v", oldNode, err)
-	}
-	newNode, ok := (objCopy).(*v1.Node)
```
Comment (sttts) [inline pkg/controller/controller_utils.go]: this was replacing the node pointer with a new object.

**Comment by kargakis** [comment]
awesome

**Pull request merged**